pkg/csi/service/service.go

```diff
@@ -95,9 +96,22 @@ func (s *service) BeforeServe(
 	// Get the SP's operating mode.
 	s.mode = csictx.Getenv(ctx, gocsi.EnvVarMode)
 
+	// Set glog level based on CSI debug being enabled
+	var glogLevel string
+	csiDebug := csictx.Getenv(ctx, gocsi.EnvVarMode)
```

Comment:
This doesn't get you the DEBUG flag, it gets you the MODE. But like I put in the example of my other comment, you don't need to go read the ENV var here, because GoCSI has already read and parsed the logger's level. Just do:

```go
glogLevel := "2"
lvl := log.GetLevel()
if lvl = log.DebugLevel {
  glogLevel = "4"
}
```

That would be my approach.

Reply:
ok will update